pkg/gardenlet/controller/federatedseed/networkpolicy/hostnameresolver/resolver_test.go

```diff
@@ -0,0 +1,177 @@
+	)
+	BeforeEach(func() {
+		updateCalled = false
+		lock = sync.Mutex{}
+		f = &fakeLookup{
+			addrs: []string{"5.6.7.8", "1.2.3.4"}, // sorting check
+		}
+		r = &resolver{
+			lookup:        f,
+			upstreamPort:  1234,
+			refreshTicker: time.NewTicker(time.Millisecond * 3),
+			log:           logrus.NewEntry(logger.NewNopLogger()),
+			onUpdate: func() {
+				lock.Lock()
+				updateCalled = true
+				cancelFunc()
+				lock.Unlock()
+			},
+		}
+	})
+
+	It("should return correct subset", func(done Done) {
+		ctx, cancelFunc = context.WithTimeout(context.Background(), time.Millisecond*2)
+
+		Expect(r.HasSynced()).To(BeFalse(), "HasSync should be false before starting")
+
+		r.Start(ctx)
+
+		Expect(r.HasSynced()).To(BeTrue(), "HasSync should be true after start")
+		Expect(updateCalled).To(BeTrue(), "update should be called")
+
+		Expect(r.Subset()).To(ConsistOf(corev1.EndpointSubset{
+			Addresses: []corev1.EndpointAddress{
+				{IP: "1.2.3.4"}, {IP: "5.6.7.8"},
+			},
+			Ports: []corev1.EndpointPort{{Protocol: corev1.ProtocolTCP, Port: 1234}},
+		}))
+
+		close(done)
+	}, 0.2)
+
+	It("should not return that it has synced", func(done Done) {
+		ctx, cancelFunc = context.WithTimeout(context.Background(), time.Millisecond*2)
+
+		Expect(updateCalled).To(BeFalse(), "update should not be called")
+		Expect(r.HasSynced()).To(BeFalse(), "HasSync should be false")
+
+		close(done)
+	}, 0.2)
+
+	It("should not return that it has synced if error occurs", func(done Done) {
+		ctx, cancelFunc = context.WithTimeout(context.Background(), time.Millisecond*2)
+
+		f.setAddrs(nil)
+		f.setErr(errors.New("some-error"))
+
+		r.Start(ctx)
+
+		Expect(updateCalled).To(BeFalse(), "update should not be called")
+		Expect(r.HasSynced()).To(BeFalse(), "HasSync should be false")
+
+		close(done)
+	}, 0.2)
+
+	It("should return correct subset after resync", func(done Done) {
+		ctx, cancelFunc = context.WithTimeout(context.Background(), time.Millisecond*10)
+		r.onUpdate = func() {
+			lock.Lock()
+			updateCalled = true
+			lock.Unlock()
+		}
+
+		Expect(r.HasSynced()).To(BeFalse(), "HasSync should be false before starting")
+
+		go r.Start(ctx)
+
+		Eventually(func() bool { //nolint:unlambda
+			return r.HasSynced()
+		}, time.Millisecond*3, time.Millisecond).Should(BeTrue(), "HasSync should be true after start")
+
+		Eventually(func() bool { //nolint:unlambda
+			lock.Lock()
+			defer lock.Unlock()
+			return updateCalled
+		}, time.Millisecond*3, time.Millisecond).Should(BeTrue(), "update should be called")
+
+		Consistently(func() []corev1.EndpointSubset { //nolint:unlambda
+			return r.Subset()
+		}, time.Millisecond*3, time.Millisecond).Should(ConsistOf(corev1.EndpointSubset{
+			Addresses: []corev1.EndpointAddress{
+				{IP: "1.2.3.4"}, {IP: "5.6.7.8"},
+			},
+			Ports: []corev1.EndpointPort{{Protocol: corev1.ProtocolTCP, Port: 1234}},
+		}))
+
+		f.setAddrs([]string{"5.6.7.8"})
+
+		Eventually(func() []corev1.EndpointSubset { //nolint:unlambda
```

Comment:
Why `//nolint:unlambda` is required and what it does? `make check` is passing when it is not present.

Reply:
This is a false-positive warning on the linter. For example visual studio code with gopls backend complains about it.